tests/admin_inlines/tests.py

```diff
@@ -854,6 +855,49 @@ def test_delete_inlines(self):
         self.assertEqual(len(self.selenium.find_elements_by_css_selector(
             'form#profilecollection_form tr.dynamic-profile_set#profile_set-2')), 1)
 
+    def test_gone_deletes(self):
+        """
+        Ensure that the delete buttons are also present in the form after
+        a form validation error. Bug #15910.
+        """
+        self.admin_login(username='super', password='secret')
+        self.selenium.get('%s%s' % (self.live_server_url,
+            reverse('admin:admin_inlines_holder2_add')))
+
+        # Add a few inlines
+        self.selenium.find_element_by_link_text('Add another Inner2').click()
+        self.selenium.find_element_by_link_text('Add another Inner2').click()
+        self.assertEqual(len(self.selenium.find_elements_by_css_selector(
+            '#inner2_set-group div.dynamic-inner2_set')), 5)
+        self.assertEqual(len(self.selenium.find_elements_by_css_selector(
```

Comment:
Please use this indent style:
```
self.assertEqual(
    len(self.selenium.find_elements_by_css_selector('form#holder2_form div.dynamic-inner2_set#inner2_set-0')
    1
)
```
(maybe a loop or helper function would make this less redundant and easier to read... or factoring "form#holder2_form div.dynamic-inner2_set#inner2_set-" into a constant so it's not repeated so much?)